File: ecom-back/services/product-service/src/main/java/com/ecom/product/model/Product.java
package com.ecom.product.model;

import java.math.BigDecimal;
import java.util.List;

import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.index.TextIndexed;
import org.springframework.data.mongodb.core.mapping.Document;

import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Document(collection = "products")
@Getter
@Setter
@NoArgsConstructor
public class Product {

    @Id
    private String id;

    @TextIndexed
    private String name;

    @TextIndexed
    private String description;

    private String category;
    private String brand;
    private BigDecimal price;
    private boolean active = true;
    private List<String> colors;
    private List<String> sizes;
}

File: ecom-storefront/lib/types/product.ts
export interface Product {
  id: string;
  name: string;
  description?: string;
  category: string;
  brand: string;
  price: number;
  active: boolean;
  colors?: string[];
  sizes?: string[];
  imageUrls?: string[];
}

export interface ProductRequest {
  name: string;
  description?: string;
  category: string;
  brand: string;
  price: number;
  colors?: string[];
  sizes?: string[];
  active?: boolean;
  imageUrls?: string[];
}

export interface ProductPage {
  content: Product[];
  totalElements: number;
  totalPages: number;
  size: number;
  number: number;
  first: boolean;
  last: boolean;
}
